floats/floats.go

```diff
@@ -923,3 +923,23 @@ func Within(s []float64, v float64) int {
 	}
 	return -1
 }
+
+// SumCompensated returns the sum of the elements of the slice calculated with greater
+// accuracy than Sum at the expense of additional computation.
+func SumCompensated(s []float64) float64 {
+	// SumCompensated uses an improved version of Kahan's compensated
+	// summation algorithm proposed by Neumaier.
+	// See https://en.wikipedia.org/wiki/Kahan_summation_algorithm for details.
+	sum := 0.
+	c := 0.
```

Comment:
`var sum, c float64`

The use of a single `.` is a small mark to denote type, so better to be explicit.